### ISSUE TITLE
Evaluate SQL fuzzy query like search performance 

### ISSUE THREAD

**Issue by enicolasgomez** [MetaCell/asu-olfactory#8]
@jrmartin if you see the normalization script here:

https://github.com/MetaCell/asu-olfactory/blob/main/normalize.py

we need to grab the normalized data en bulk insert it into a POSTGRESQL instance running locally. with a single table having CID (numeric), Synonym (varchar) in which both columns will be indexes.

when the process is finished it will populate around 35M rows, we need to evaluate performance.

summarizing:

- start up a postgresql instance with 1 database and create 1 table with the given description
- download the dataset:
https://ftp.ncbi.nlm.nih.gov/pubchem/Compound/Extras/CID-Synonym-unfiltered.gz
- run normalize.py on it
- this will create around 600 files with 50.000 rows
- check that the normalization worked as exepcted with some random sampling (manually)
- create a bulk_insert.py script that would go over the files and bulk insert into the newly created table

finally:

- evaluate LIKE '%[value]%' kind of queries performance
- if is not good enough, we'll need to evaluate configurations that would avoid table scan

Here's a description on how data is expected to be normalized:

https://app.zenhub.com/workspaces/asu-olfactory-625d568a637ec8001a4e40ac/issues/metacell/asu-olfactory/1

**Comment by enicolasgomez** [comment]
here's a good POC:

https://www.cybertec-postgresql.com/en/postgresql-more-performance-for-like-and-ilike-statements/

50 M rows taking 4.5 secs for a LIKE %value% kind of query which after some indexing magic turns to 75ms for a full md5 hash

**Comment by jrmartin** [comment]
@enicolasgomez I run the normalize.py script, and it created 3147 CSV files for me. 
Data seems correct 
![image.png](https://images.zenhubusercontent.com/5b8f86ce39827c509079f60a/08dd3f54-686f-426f-80a5-8e64d5d4b9c8)

**Comment by enicolasgomez** [comment]
@jrmartin added
https://github.com/MetaCell/asu-olfactory/blob/main/querypoc.py
to test the approach which is valid, took 39 ms to query a md5 hash between 50M

**Comment by jrmartin** [comment]
@enicolasgomez 

I created an AWS Linux Instance to test my scripts. I created a script that populates the DB Table with the CSV normalized data. 

I created two Tables, one with the GIN implementation and one without it. To test differences in performance.

It took 4 minutes to populate the Table with the CSV normalized data.
It took 28.5 minutes to perform the GIN indexing. 
In total, the process of population the Table with GIN indexing takes 32.5 minutes.

This is the performance searching in the GIN indexed Table and regular Table without GIN indexing. As expected, GIN is faster.
![image.png](https://images.zenhubusercontent.com/5b8f86ce39827c509079f60a/4faf2eaa-34bd-439a-91c3-8485d4bc5126)
